logging.md

```diff
@@ -0,0 +1,39 @@
+# Logging
+
+We want traceability from HTTP calls.
+- Each HTTP call needs to generate a call ID (preferable on the proxy). This ID must be should be pass to the scope of the ZIO application. So every log will mention this ID. This ID must also be returned to the user as a HTTP header.
+- In case of errors and problem reports those IDs can be use by the user to get support.
+
+Level 3 support can be provided to user during the logging retention policies. We user need to provide the value in the header `X-Request-Id` to get this support.
+
+## Annotates
+
+We have special annotations in the log so there is traceability between the different logs. 
+Here is the list of annotations and their meaning that we currently have:
+
+- `request-id` - Is the HTTP header `X-Request-Id` from the caller.
+  - If this header is missing, it is create by the APISIX https://apisix.apache.org/docs/apisix/plugins/request-id/. See the configuration how to enable in the file [apisixroute.yaml](./infrastructure/charts/agent/templates/apisixroute.yaml)
+
+## Code
+
+To have a concise code we have created a Middleware that modifies the Annotations in the Scope before each execution of that endpoint, to include the trace ID of the request.
+See code in file `LogUtils.scala`.
+
+## Logging Backend
+
+We use the Simple Logging Facade for Java (SLF4J) API to call the logging backend is determined at runtime.
+
+### Logging Pattern
```

Comment:
Will the colouring cause any issues in the logging infrastructure? I know it's very useful for the console but if it ends up as extra text in Grafana - it may cause problems - something we can test locally perhaps

Reply:
Tested locally - looks great in Grafana :-)

![image](https://github.com/hyperledger-labs/open-enterprise-agent/assets/109518299/5688faab-7930-49f1-9ac5-f2e58af0b9af)